packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -668,12 +721,21 @@ function mountClassInstance(
   }
 
   const instance = workInProgress.stateNode;
-  const unmaskedContext = getUnmaskedContext(workInProgress, ctor, true);
-
   instance.props = newProps;
   instance.state = workInProgress.memoizedState;
   instance.refs = emptyRefsObject;
-  instance.context = getMaskedContext(workInProgress, unmaskedContext);
+
+  const contextType = ctor.contextType;
+  if (
+    typeof contextType === 'object' &&
+    contextType !== null &&
+    typeof contextType.unstable_read === 'function'
```

Comment:
Do we need this extra check? Seems like we can just throw.

Reply:
I guess we don't, since we also [provide a nice DEV mode warning if it's not a function](https://github.com/facebook/react/blob/0dc0ddc1ef5f90fe48b58f1a1ba753757961fc74/packages/react-reconciler/src/ReactFiberClassComponent.js#L355-L368). I'll remove this type check.

Reply:
Actually, if we don't check in at least the constructor path, we'll throw _before_ our DEV warning. So I think it's best to leave these checks in place after all. If you feel strongly about it, let's talk and come up with another plan. I think the DEV warnings are useful though.